### PR TITLE
Weather app

### LINKED ISSUE #9: Get the weather forecast
Get weather forecast based on user location

### PR DESCRIPTION
**Pull request by Patelrahul4884**
add utils.py file in Weather app that fetches the weather using geo coordinate(latitude & longitude)

### CODE REVIEW

**Comment by HealYouDown** [comment]
- [x] There are some conflicts that you have to resolve.
- [x] As I mentioned in Discord, please check your code again with a linter enabled and fix these issues.
- [x] You will also store the key for your API not inside your function. It should be in the config and then imported inside your `utils.py` file.
- [x] Please also delete all files that are not used. (`admin.py` and `tests.py` for example).
- [x] You also pushed your vscode settings folder, you have to delete that one. (I recommend to create a global gitignore file that excludes it)
- [x] Please also use type hints for your function arguments as well as the return type.
- [ ] We still need a proxy model for the weather. Have a look at the models inside the dashboard folder how it works.

I told you to use the requests module, as it is a bit easier to do the fetching, but it's not the end of the world that you used urllib.

- [x] However, we need the following returned from the function. 
```py
{
	"weather": {
		"celsius": float,
		"fahrenheit": float,
	},
	"city": str,
	"sunset": datetime,
	"sunrise": datetime
}
```

- [x] Please also remove your code block at the bottom after testing the functionallity for your file. It should not be pushed to the master branch.

File: concerned-coyotes/earlyinternet/weather/utils.py [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
+    # list_of_data contain json encoded content
+    list_of_data = source.json()
+    data = {"weather": {
+        "celsius": int(list_of_data['main']['temp'])-272,
+        "fahrenheit": int(((list_of_data['main']['temp'])-272)*1.8+32)},
+        "city": str(list_of_data['name']),
+        "sunset": datetime.fromtimestamp(
+            list_of_data['sys']['sunset']),
+        "sunrise": datetime.fromtimestamp(
+            list_of_data['sys']['sunset']),
```
Comment (maxrdu) [inline concerned-coyotes/earlyinternet/weather/utils.py]: I assume it should be `list_of_data['sys']['sunrise']`

File: concerned-coyotes/earlyinternet/weather/utils.py [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
+        }
     return data
 
-if __name__=="__main__":
-    weather_data=get_weather(21.178,72.83)
-    print(weather_data)
+
+if __name__ == "__main__":
+    weather_data = get_weather(21.178, 72.83)
```
Comment (maxrdu) [inline concerned-coyotes/earlyinternet/weather/utils.py]: Please remove this

File: concerned-coyotes/earlyinternet/weather/utils.py [hunk 3]
```diff
@@ -1,21 +1,29 @@
...
+    """
+    Take latitude and longitude to get weather data
+    using openweather api and return data in dict format
+    """
+    weather_key = '0a2c03e9dee2533389d7b459d0f8595e'
+    # source contain JSON data from API
+    source = requests.get(
+        f'http://api.openweathermap.org/data/2.5/weather?lat={lat}&lon={lon}&appid={weather_key}')
+    # list_of_data contain json encoded content
+    list_of_data = source.json()
```
Comment (maxrdu) [inline concerned-coyotes/earlyinternet/weather/utils.py]: Just `data` as a name is sufficient

File: concerned-coyotes/earlyinternet/weather/utils.py [hunk 4]
```diff
@@ -1,21 +1,29 @@
...
+    data = {"weather": {
+        "celsius": int(list_of_data['main']['temp'])-272,
+        "fahrenheit": int(((list_of_data['main']['temp'])-272)*1.8+32)},
```
Comment (maxrdu) [inline concerned-coyotes/earlyinternet/weather/utils.py]: The temp value is Fahrenheit by default no need to convert to celsius and then back to fahrenheit

**Review by maxrdu — CHANGES_REQUESTED** [review]

File: concerned-coyotes/earlyinternet/earlyinternet/urls.py [hunk 5]
```diff
@@ -14,8 +14,9 @@
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
 from django.contrib import admin
-from django.urls import path
+from django.urls import path,include
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    path('',include('weather.urls'))
 ]
```
Comment (maxrdu) [inline concerned-coyotes/earlyinternet/earlyinternet/urls.py]: Weather currenlty has no views so you dont need to add its (nonexistent) urls

**Review by maxrdu — CHANGES_REQUESTED** [review]

File: concerned-coyotes/earlyinternet/weather/utils.py (hunk 4 above)
Comment (Patelrahul4884) [reply concerned-coyotes/earlyinternet/weather/utils.py]: I think the default is kelvin that's why I convert it

File: concerned-coyotes/earlyinternet/earlyinternet/urls.py (hunk 5 above)
Comment (Patelrahul4884) [reply concerned-coyotes/earlyinternet/earlyinternet/urls.py]: I don't have this file in my editor

File: concerned-coyotes/earlyinternet/weather/utils.py (hunk 4 above)
Comment (maxrdu) [reply concerned-coyotes/earlyinternet/weather/utils.py]: You are right my bad

**Review by maxrdu — APPROVED** [review]

**Pull request merged**